lib/cartopy/mpl/geoaxes.py

```diff
@@ -1542,26 +1540,11 @@ def scatter(self, *args, **kwargs):
             A :class:`~cartopy.crs.Projection`.
 
         """
-        t = kwargs.get('transform', None)
-        # Keep this bit - even at mpl v1.2
-        if t is None:
-            t = self.projection
-        if hasattr(t, '_as_mpl_transform'):
-            kwargs['transform'] = t._as_mpl_transform(self)
-
-        # exclude Geodetic as a valid source CS
-        if (isinstance(kwargs.get('transform', None),
-                       InterProjectionTransform) and
-                kwargs['transform'].source_projection.is_geodetic()):
```

Comment:
This looks like a different check, no?

Reply:
Yes, I was not even sure it was actually possible to enter this statement because above you're setting kwargs['transform'] to a mpl transform...

Also note that below it actually mentions `contouring` but this is in `scatter`. There are a lot of inconsistencies here that could be addressed later.

I've added it back in for now and changed contouring to scatter.

Reply:
True, I'm not sure how you might end up with an `InterProjectionTransform`, but I guess someone could pass it in directly.